api/handlers/user.go

```diff
@@ -83,3 +84,29 @@ func UserInfo(c *gin.Context) {
 		},
 	})
 }
+
+func SignUp(c *gin.Context) {
+	var req struct {
+		Email    string `json:"email"`
+		Password string `json:"password"`
+		FullName string `json:"fullName"`
+	}
```

Comment:
I think an open signup might be dangerous. Perhaps we can have a pre-shared key here to prevent abuse (optional - via environment variable)